beluga_system_tests/test/test_system_new.cpp

```diff
@@ -170,13 +173,10 @@ auto particle_filter_test(
         probability_estimator.reset();
       }
 
-      auto make_random_particle = [&sensor] {
-        static thread_local auto engine = std::mt19937{std::random_device()()};
-        return beluga::make_from_state<Particle>(sensor.make_random_state(engine));
-      };
-
       particles |= beluga::views::sample |
-                   beluga::views::random_intersperse(make_random_particle, random_state_probability) |
+                   beluga::views::random_intersperse(
+                       ranges::compose(beluga::make_from_state<Particle>, std::ref(map_distribution)),
+                       random_state_probability) |
```

Comment:
meta: We can make `random_intersperse` smarter so we can write this:

```suggestion
                   beluga::views::random_intersperse(std::ref(map_distribution), random_state_probability) |
```

I'm not super convinced that we should though.

Reply:
+1 to having `beluga::views::random_intersperse` accept a distribution instead of a callable. It's not a substantial detour, API wise, and for a library of Monte Carlo methods, pretty much everything is a distribution.

Reply:
Hmm, I think `beluga::views::random_intersperse` should take callables. We can also take distributions, but the thing here is that those distributions/callables should generate values that are compatible with the input range.

In this case, `beluga::views::sample` is generating particles (as it should) since it's sampling from a particle range.
The question is, should we make `beluga::views::random_intersperse` check if the callable does not generate particles but state values compatible with the input particles and call `beluga::make_from_state` internally?

Reply:
I like the current state of affairs. No need for `random_intersperse` to do more than it should.